telemetry/telemetry_linux.go

```diff
@@ -88,14 +88,19 @@ func (report *Report) GetSystemDetails() {
 
 // This function  creates a report with os details(ostype, version).
 func (report *Report) GetOSDetails() {
-	linesArr, err := ReadFileByLines("/etc/issue")
+	linesArr, err := ReadFileByLines("/etc/os-release")
 	if err != nil || len(linesArr) <= 0 {
 		report.OSDetails = &OSInfo{OSType: runtime.GOOS}
-		report.OSDetails.ErrorMessage = "reading /etc/issue failed with" + err.Error()
+		report.OSDetails.ErrorMessage = "reading /etc/os-release failed with" + err.Error()
 		return
 	}
 
-	osInfoArr := strings.Split(linesArr[0], " ")
+	osInfoArr := make(map[string]string)
+
+	for i := range linesArr {
+		s := strings.Split(linesArr[i], "=")
```

Comment:
can we check if split succeeds by checking the length of s?

Reply:
Please check if the latest commit is what you want.